frontend/src/components/assessments/student-experience/SubmitButton.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useContext, useEffect, useMemo } from "react";
+import { useMutation } from "@apollo/client";
+import { Button } from "@chakra-ui/react";
+
+import { SUBMIT_TEST } from "../../../APIClients/mutations/TestSessionMutations";
+import AuthContext from "../../../contexts/AuthContext";
+import StudentContext from "../../../contexts/StudentContext";
+import WriteAssessmentContext from "../../../contexts/WriteAssessmentContext";
+import {
+  isCompleted,
+  mapAnswersToResultsArray,
+} from "../../../utils/StudentUtils";
+import Modal from "../../common/Modal";
+import Toast from "../../common/Toast";
+
+const SubmitButton = (): React.ReactElement => {
+  const { testSession } = useContext(StudentContext);
+  const { answers, setIsSubmitted, setIsLoading } = useContext(
+    WriteAssessmentContext,
+  );
+  const { showToast } = Toast();
+  const { authenticatedUser } = useContext(AuthContext);
+  const [showConfirmModal, setShowConfirmModal] = React.useState(false);
+
+  const [submitTest, { loading, error }] = useMutation<{
+    submitTest: string;
+  }>(SUBMIT_TEST);
+
+  useEffect(() => {
+    if (loading) {
+      setIsLoading(true);
+    }
+  }, [loading, setIsLoading]);
+
+  const handleSubmitTest = async () => {
+    await submitTest({
+      variables: {
+        id: testSession?.id,
+        result: {
+          student: authenticatedUser?.id,
+          answers: mapAnswersToResultsArray(answers),
+        },
+      },
+    });
+    if (error) {
+      setIsLoading(false);
+      showToast({
+        message: "Assessment failed to submit. Please try again.",
+        status: "error",
+      });
+    } else {
+      setIsLoading(false);
+      setIsSubmitted(true);
+    }
+  };
+
+  const incompleteQuestionCount = useMemo(() => {
+    return answers.filter((answer) => !isCompleted(answer)).length;
+  }, [answers]);
+
+  const header = useMemo(() => {
+    const question = incompleteQuestionCount === 1 ? "Question" : "Questions";
+    if (incompleteQuestionCount) {
+      return `You have ${incompleteQuestionCount} Unanswered ${question}`;
+    }
+    return "Submit the test";
+  }, [incompleteQuestionCount]);
```

Comment:
a couple notes:
- in the designs the `${incompleteQuestionCount} Unanswered` text is red and bolded, any thoughts on the most elegant way to do this? it's kind of hard to adapt the existing modal component, but i'm hesitant to create a modal from scratch just for this as well
- in the designs, it shows the word version of the number, not the number itself (i.e `one` instead of `1`). is there a easy way to do this? i can't think of one... unless i'm overlooking something obvious

Reply:
For 1., you could probably just change the type of `header` (and `body`) to `React.ReactNode`, and pass in a React Element instead. i.e., you'd return something like `<>You have {incompleteQuestionCount} <Box color="red">Unanswered</Box>{question}</>;` (the other `return` could stay unchanged).

For 2., yeah idk, you could add a library for this if you wanted or just manually check, but I'd probably just use the # unless the designer cares.

Reply:
oh awesome sounds good! thank you 😊